hbase-common/src/main/java/org/apache/hadoop/hbase/io/encoding/PrefixKeyDeltaEncoder.java

```diff
@@ -48,8 +48,9 @@
 public class PrefixKeyDeltaEncoder extends BufferedDataBlockEncoder {
 
   @Override
-  public int internalEncode(Cell cell, HFileBlockDefaultEncodingContext encodingContext,
+  public void internalEncode(Cell cell, HFileBlockDefaultEncodingContext encodingContext,
```

Comment:
These 3 encoders basically have an Abstract parent class..  See this logic of updating the size can be moved there..  The internalEncode can return the encodedSize as before..  And abstract class uses that. And then it can do math of out.size () based too and update in context

Reply:
Looking into this. Will see if we can add this logic in the encode method of the abstract parent from where `internalEncode()` is called.